src/settings/config.rs

```diff
@@ -1,97 +1,301 @@
+        Ok(match &s.to_lowercase()[..] {
+            "emacs" => EditMode::Emacs,
+            "vi" => EditMode::Vi,
+            _ => return Err(usage_error(format!("{} is not a valid edit mode", s))),
+        })
+    }
+}
+
 impl Default for EditMode {
     fn default() -> Self {
         EditMode::Emacs
     }
 }
 
+impl TryFrom<&OwnedStruct> for ShellConfig {
+    type Error = ShellError;
+
+    fn try_from(value: &OwnedStruct) -> Result<Self, Self::Error> {
+        let mut config = ShellConfig::default();
+
+        if let Some(elem) = value.get("ui") {
+            let ui = elem
+                .as_struct()
+                .ok_or(usage_error("`ui` should be a struct"))?;
+            config.ui = UiConfig::try_from(ui)?;
+        }
+
+        if let Some(elem) = value.get("debug") {
+            let debug = elem
+                .as_struct()
+                .ok_or(usage_error("`debug` should be a struct"))?;
+            config.debug = DebugConfig::try_from(debug)?;
+        }
+
+        if let Some(elem) = value.get("default_ledger") {
+            config.default_ledger = Some(
+                elem.as_str()
+                    .ok_or(usage_error("`default_ledger` should be a string"))?
+                    .to_string(),
+            );
+        }
+
+        if let Some(elem) = value.get("ledgers") {
+            let ledgers = elem
+                .as_sequence()
+                .ok_or(usage_error("`ledgers` should be a list"))?;
+            let deser: Result<Vec<_>, ShellError> = ledgers
+                .iter()
+                .map(|elem| {
+                    let ledger = elem
+                        .as_struct()
+                        .ok_or(usage_error("`ledgers` should be a list of structs"))?;
+                    LedgerConfig::try_from(ledger)
+                })
+                .collect();
+            let deser = deser?;
+            if !deser.is_empty() {
+                config.ledgers = Some(deser);
+            }
+        }
+
+        Ok(config)
+    }
+}
+
+impl TryFrom<&OwnedStruct> for UiConfig {
+    type Error = ShellError;
+
+    fn try_from(value: &OwnedStruct) -> Result<Self, Self::Error> {
+        let mut ui = UiConfig::default();
+
+        ui.auto_commit = if let Some(elem) = value.get("auto_commit") {
+            elem.as_bool()
+                .ok_or(usage_error("`ui.auto_commit` should be a bool"))?
+        } else {
+            true
+        };
+
+        if let Some(elem) = value.get("prompt") {
+            ui.prompt = Some(
+                elem.as_str()
+                    .ok_or(usage_error("`ui.prompt` should be a string"))?
+                    .to_string(),
+            );
+        };
+
+        ui.format = if let Some(elem) = value.get("format") {
+            let format = elem
+                .as_str()
+                .ok_or(usage_error("`ui.format` should be a string"))?;
+            FormatMode::from_str(format)?
+        } else {
+            FormatMode::Ion
```

Comment:
Is the `ui.format` already `Ion` after `let mut ui = UiConfig::default();`? Is the `else` logic needed here?

Reply:
That's a good point! I'll update this and all the other places.